content/chainguard/chainguard-enforce/how-to-install-chainctl.md

```diff
@@ -84,17 +84,231 @@ You should receive output similar to the following.
+
+For completeness, you can use the following script to check `chainctl` binaries using Cosign. It is a combination of the previous steps into one script with some additional checks for missing tools, and error handling.
+
+```bash
+#!/bin/bash
+set -Eeuo pipefail
+export COSIGN_EXPERIMENTAL=1
+
+trap 'cmd_error $LINENO "${BASH_COMMAND}"' ERR
+function cmd_error() {
+  echo "error running command on line $1"
+  echo "command is: $2"
+}
+
+function check_signature {
+    chainctl_version=$(chainctl version 2>&1 |awk '/GitVersion/ {print $2}')
+    chainctl_os_arch="chainctl_$(uname -s | tr '[:upper:]' '[:lower:]')_$(uname -m)"
+    sig_url="https://dl.enforce.dev/chainctl/${chainctl_version?}/${chainctl_os_arch?}.sig"
+    cert_url="https://dl.enforce.dev/chainctl/${chainctl_version?}/${chainctl_os_arch?}.cert.pem"
+    cosign verify-blob --signature "${sig_url?}" --cert "${cert_url?}" "$(command -v chainctl)"
+}
+
+function check_executable {
+    status=0
+    executable_name="${1?}"
+    command -v "${executable_name?}" > /dev/null || status=$?
+    if [[ "${status?}" -ne 0 ]]; then
+        echo "${executable_name?} not found"
+        exit 1
+    fi
+}
+
+check_executable "cosign"
+check_executable "chainctl"
+check_signature
+```
+
+#### Verifying `chainctl` using the Rekor Transparency Log
+
+You can look up the Rekor entry for your version of `chainctl` by searching the log for the SHA256 hash of `chainctl`. You can use the `rekor-cli` tool or `curl` to find matching Rekor entries. If you need to install `rekor-cli`, follow our guide [How to Install the Rekor CLI](https://edu.chainguard.dev/open-source/sigstore/rekor/how-to-install-rekor/). When using either tool, ensure that you have the `jq` utility installed so that you can parse their output.
+
+To search Rekor, set a shell variable to the SHA256 hash of your `chainctl` binary:
+
+```sh
+SHASUM=$(shasum -a 256 $(which chainctl) |awk '{print $1}')
+```
+
+If you are using the `rekor-cli` client, search for the hash with the following command:
+
+```sh
+rekor-cli search --sha "${SHASUM?}"
+```
+
+If you are using `curl`, run the following:
+
+```sh
+curl -X POST -H "Content-type: application/json" 'https://rekor.sigstore.dev/api/v1/index/retrieve' --data-raw "{\"hash\":\"sha256:$SHASUM\"}"
+```
+
+If there is an entry for your version of `chainctl` you will receive output like the following:
+
+```
+# rekor-cli output
+Found matching entries (listed by UUID):
+24296fb24b8ad77a1a481ce6eeee6306dffbba9fef701a21623ca0b07780b32b009395c905f7df7a
+
+# curl output
+["24296fb24b8ad77a1a481ce6eeee6306dffbba9fef701a21623ca0b07780b32b009395c905f7df7a"]
+```
+
+Set a shell variable called `UUID` to the returned entry:
+
+```sh
+UUID="24296fb24b8ad77a1a481ce6eeee6306dffbba9fef701a21623ca0b07780b32b009395c905f7df7a"
+```
+
+Now you can use the returned UUID to retrieve the associated Rekor log entry. If you are using `rekor-cli` run the following:
+
+```sh
+rekor-cli get --uuid "${UUID?}"
```

Comment:
I needed to update Rekor for this to work, maybe we should say that people should have a minimum version of v1.0.0 ?